cmake/extensions.cmake

```diff
@@ -2282,6 +2282,29 @@ Relative paths are only allowed with `-D${ARGV1}=<path>`")
   endif()
 endfunction()
 
+# Usage:
+#   zephyr_file_rename(<name> <newname>)
```

Comment:
zephyr_file_maybe_rename
zephyr_file_rename_if_changed

We need to communicate that the file is not always renamed, or else the code looks really weird.

```
touch foo.txt.new
mv foo.txt.new foo.txt
```

Reply:
perhaps

zephyr_file_rename_only_if_different

to be consistent with CMake.

Reply:
> We need to communicate that the file is not always renamed, or else the code looks really weird.

I thought about it, but actually, how will you tell whether or not the file was actually renamed if the content is identical ?
Especially if we decided to adjust the timestamp back in time.

But with your comment here: https://github.com/zephyrproject-rtos/zephyr/pull/42532#discussion_r801466010

how about calling the function:
```
zephyr_file_copy(<oldname> <newname> [ONLY_IF_DIFFERENT])
```
so we support `ONLY_IF_DIFFERENT` as optional argument.

That way, when CMake minimum required is someday bumped >=3.21 we can simply swap all occurrences of:
`zephyr_file_copy(foo.txt bar.txt ONLY_IF_DIFFERENT)` to `file(COPY_FILE foo.txt bar.txt ONLY_IF_DIFFERENT)`

and then followed with a:
```
file(REMOVE foo.txt)
```

That will remove any ambiguity in command name / arguments.

Reply:
I'd prefer

zephyr_file_copy_file_only_if_different

to avoid the optional argument that we always intend to use AFAICT.

Reply:
in this particular PR we always intend to use `ONLY_IF_DIFFERENT`, but because `file(COPY_FILE <oldname> <newname>)` is new in 3.21 and Zephyr has 3.20 as minimum required, I do see the value in having a pure file copy available for general use.

Having the functionality of `file(COPY_FILE ...)` available under a zephyr wrapper makes a lot of sense as long as we support CMake 3.20.

Note that the existing CMake `file(COPY <file>... DESTINATION <dir>)` doesn't allow to copy to a different name.

Reply:
all right then

Reply:
done